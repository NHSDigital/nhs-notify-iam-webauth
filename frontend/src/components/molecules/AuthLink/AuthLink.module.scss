@use 'nhsuk-frontend/packages/core/settings/colours' as nhsuk;

.auth-link {
  &__link {
    color: nhsuk.$color_nhsuk-white;
    padding: 1em;

    &:visited {
      color: nhsuk.$color_nhsuk-white;
    }
  }
}
