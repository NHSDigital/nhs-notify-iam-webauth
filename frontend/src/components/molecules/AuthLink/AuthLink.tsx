'use client';

import React from 'react';
import { useAuthenticator } from '@aws-amplify/ui-react';
import content from '@/src/content/content';
import styles from './AuthLink.module.scss';
import { authenticatorSelector } from '../../../utils/authenticator-selector';

export const AuthLink = () => {
  const { authStatus } = useAuthenticator(authenticatorSelector);

  let id = 'sign-in-link';
  let linkContent = content.components.headerComponent.links.signIn;
  if (authStatus === 'authenticated') {
    id = 'sign-out-link';
    linkContent = content.components.headerComponent.links.signOut;
  }

  return (
    <div className={styles['auth-link']} data-testid='auth-link'>
      <a id={id} className={styles['auth-link__link']} href={linkContent.href}>
        {linkContent.text}
      </a>
    </div>
  );
};
