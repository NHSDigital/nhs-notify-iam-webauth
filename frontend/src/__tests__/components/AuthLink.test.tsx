import { mockDeep } from 'jest-mock-extended';
import { UseAuthenticator, useAuthenticator } from '@aws-amplify/ui-react';
import { render, screen } from '@testing-library/react';
import { AuthSession } from '@aws-amplify/auth';
import { AuthLink } from '@/src/components/molecules/AuthLink/AuthLink';

jest.mock('@aws-amplify/ui-react');
jest.mock('aws-amplify/auth', () => ({
  fetchAuthSession: async () =>
    mockDeep<AuthSession>({
      tokens: {
        idToken: {
          payload: {
            email: 'test-email',
          },
        },
      },
    }),
}));

const useAuthenticatorMock = jest.mocked(useAuthenticator);

test('AuthLink - authenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'authenticated',
    })
  );

  const container = render(<AuthLink />);

  await screen.findByText('Sign out');

  expect(container.asFragment()).toMatchSnapshot();
});

test('AuthLink - unauthenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'unauthenticated',
    })
  );

  const container = render(<AuthLink />);

  await screen.findByText('Sign in');

  expect(container.asFragment()).toMatchSnapshot();
});

test('AuthLink - configuring', async () => {
  jest.mocked(useAuthenticator).mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'configuring',
    })
  );

  const container = render(<AuthLink />);

  await screen.findByText('Sign in');

  expect(container.asFragment()).toMatchSnapshot();
});
