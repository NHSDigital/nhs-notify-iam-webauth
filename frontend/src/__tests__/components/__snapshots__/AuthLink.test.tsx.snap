// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLink - authenticated 1`] = `
<DocumentFragment>
  <div
    class="auth-link"
    data-testid="auth-link"
  >
    <a
      class="auth-link__link"
      href="/auth/signout"
      id="sign-out-link"
    >
      Sign out
    </a>
  </div>
</DocumentFragment>
`;

exports[`AuthLink - configuring 1`] = `
<DocumentFragment>
  <div
    class="auth-link"
    data-testid="auth-link"
  >
    <a
      class="auth-link__link"
      href="/auth"
      id="sign-in-link"
    >
      Sign in
    </a>
  </div>
</DocumentFragment>
`;

exports[`AuthLink - unauthenticated 1`] = `
<DocumentFragment>
  <div
    class="auth-link"
    data-testid="auth-link"
  >
    <a
      class="auth-link__link"
      href="/auth"
      id="sign-in-link"
    >
      Sign in
    </a>
  </div>
</DocumentFragment>
`;
